### PR TITLE
Initialise PySpark using hooks

### PR DESCRIPTION
**Pull request by SajidAlamQB**
Signed-off-by: SajidAlamQB <90610031+SajidAlamQB@users.noreply.github.com>

## Motivation and Context

Currently PySpark is initialised in kedro using a [custom context](https://github.com/kedro-org/kedro-starters/blob/main/pyspark/%7B%7B%20cookiecutter.repo_name%20%7D%7D/src/%7B%7B%20cookiecutter.python_package%20%7D%7D/context.py). With the introduction of `after_context_created` hook, it has now become the more appropriate way to initialize PySpark.

related Issue: https://github.com/kedro-org/kedro/issues/1563

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Assigned myself to the PR
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by noklam — COMMENTED** [review]
LGTM! Don't forget to also update `kedro`'s doc for this.

File: pyspark-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/settings.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 from the Kedro defaults. For further information, including these default values, see
 https://kedro.readthedocs.io/en/stable/kedro_project_setup/settings.html."""
-from {{cookiecutter.python_package}}.context import ProjectContext
 
 # Instantiated project hooks.
-# from {{ cookiecutter.python_package }}.hooks import ProjectHooks
-# HOOKS = (ProjectHooks(),)
+from {{cookiecutter.python_package}}.hooks import SparkHooks
+
+HOOKS = (SparkHooks(),)
```
Comment (AntonyMilneQB) [inline pyspark-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/settings.py:8]:
```suggestion
from {{cookiecutter.python_package}}.hooks import SparkHooks
HOOKS = (SparkHooks(),)
```
(extremely pedantic I know 😆)

File: pyspark/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/settings.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 from the Kedro defaults. For further information, including these default values, see
 https://kedro.readthedocs.io/en/stable/kedro_project_setup/settings.html."""
-from {{cookiecutter.python_package}}.context import ProjectContext
 
 # Instantiated project hooks.
-# from {{cookiecutter.python_package}}.hooks import ProjectHooks
-# HOOKS = (ProjectHooks(),)
+from {{cookiecutter.python_package}}.hooks import SparkHooks
+
+HOOKS = (SparkHooks(),)
```
Comment (AntonyMilneQB) [inline pyspark/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/settings.py:8]:
```suggestion
from {{cookiecutter.python_package}}.hooks import SparkHooks
HOOKS = (SparkHooks(),)
```

**Review by AntonyMilneQB — APPROVED** [review]
This is perfect, thank you! I just have one question about testing - will slack you 🙂

File: pyspark-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/settings.py (hunk 1 above)
Comment (SajidAlamQB) [reply pyspark-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/settings.py]: Linters seem to not like this change 😅

Comment (AntonyMilneQB) [reply pyspark-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/settings.py:8]: Ah yes, ok. Don't do it then 😀

**Review by noklam — APPROVED** [review]

**Pull request merged**